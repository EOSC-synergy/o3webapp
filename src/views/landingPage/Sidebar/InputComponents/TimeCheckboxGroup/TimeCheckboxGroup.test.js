import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux'
import TimeCheckBoxGroup from './TimeCheckboxGroup';
import { NUM_MONTHS } from '../../../../../utils/constants';
import { createTestStore } from "../../../../../store/store"

let store
beforeEach(() => {
    store = createTestStore();
});

describe("tests basic rendering", () => {

    it('renders without crashing', () => {
        render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}} />
            </Provider>
            </>)
    }) 
    
    it('renders correctly', () => {
        
        const { container } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}} />
            </Provider>
            </>);
        
        expect(container).toMatchSnapshot();
    })
    
    it('renders all checkbox groups correctly', () => {
        
        const { getByTestId } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}} />
            </Provider>
            </>);
        
        for (let i = 0; i < (NUM_MONTHS); i++) {
            expect(getByTestId("CheckboxMonth" + (i + 1))).toBeInTheDocument();
        }
    
        expect(getByTestId("CheckboxAllYear")).toBeInTheDocument();
        
    })

});

describe("tests redux functionality", () => {
    it('selects a month correctly', () => {
        const { getByTestId } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}} />
            </Provider>
            </>);
        
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 12]); // assert: if it fails the initial state may have changed
        fireEvent.click(getByTestId("CheckboxMonth2"));
        expect(store.getState().plot.generalSettings.months).toEqual([1, 12]);
        fireEvent.click(getByTestId("CheckboxMonth2"));
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 12]);
        
    })
    
    it('selects a season correctly', () => {
        
        const { getByTestId } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}} />
            </Provider>
            </>);
        
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 12]); // assert: if it fails the initial state may have changed
    })
    
    it('selects the "All Year" checkbox correctly', () => {
        
        const { getByTestId } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}}/>
            </Provider>
            </>);
        
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 12]); // assert: if it fails the initial state may have changed
        fireEvent.click(getByTestId("CheckboxAllYear"));
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]);  
    })
});

describe("tests error handling", () => {
    it('making an incorrect selection makes an error message appear', () => {
        const { getByTestId, container } = render(<>
            <Provider store={store}>
                <TimeCheckBoxGroup reportError={() => {}}/>
            </Provider>
            </>);
        const errorMessage = "No month was selected. Data can only be fetched if at least one month is selected.";
    
        expect(store.getState().plot.generalSettings.months).toEqual([1, 2, 12]); // assert: if it fails the initial state may have changed
        fireEvent.click(getByTestId("CheckboxSeasonNum0"));
        expect(container).toHaveTextContent(errorMessage);
    
        fireEvent.click(getByTestId("CheckboxMonth4")); // select
        expect(container).not.toHaveTextContent(errorMessage);
        fireEvent.click(getByTestId("CheckboxMonth4")); // deselect
        expect(container).toHaveTextContent(errorMessage);
    
        fireEvent.click(getByTestId("CheckboxSeasonNum3"));
        expect(store.getState().plot.generalSettings.months).toEqual([9, 10, 11]);
    });
});