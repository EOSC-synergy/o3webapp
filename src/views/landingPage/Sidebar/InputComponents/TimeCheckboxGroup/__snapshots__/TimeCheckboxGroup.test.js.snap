// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests basic rendering renders correctly 1`] = `
<div>
  <div
    class="MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren css-1ws8119-MuiDivider-root"
    role="separator"
  >
    <span
      class="MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        TIME
      </p>
    </span>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item css-1h16bbz-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <label
          class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
        >
          <span
            class="MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-indeterminate MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          >
            <input
              class="PrivateSwitchBase-input css-1m9pwf3"
              data-indeterminate="true"
              data-testid="CheckboxAllYear"
              type="checkbox"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="IndeterminateCheckBoxIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-2 10H7v-2h10v2z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
          >
            All year
          </span>
        </label>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item css-1h16bbz-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              data-test-id="CheckboxSeason0"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="CheckboxSeasonNum0"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      Winter
                    </span>
                  </label>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-agnjr6"
                  >
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          checked=""
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth12"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Dec
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          checked=""
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth1"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Jan
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root Mui-checked css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          checked=""
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth2"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Feb
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              data-test-id="CheckboxSeason1"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="CheckboxSeasonNum1"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      Spring
                    </span>
                  </label>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-agnjr6"
                  >
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth3"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Mar
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth4"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Apr
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth5"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        May
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item css-1f064cs-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              data-test-id="CheckboxSeason2"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="CheckboxSeasonNum2"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      Summer
                    </span>
                  </label>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-agnjr6"
                  >
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth6"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Jun
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth7"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Jul
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth8"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Aug
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
          >
            <div
              data-test-id="CheckboxSeason3"
            >
              <div
                class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <label
                    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                  >
                    <span
                      class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                    >
                      <input
                        class="PrivateSwitchBase-input css-1m9pwf3"
                        data-indeterminate="false"
                        data-testid="CheckboxSeasonNum3"
                        type="checkbox"
                      />
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="CheckBoxOutlineBlankIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </span>
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                    >
                      Autumn
                    </span>
                  </label>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-agnjr6"
                  >
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth9"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Sep
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth10"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Oct
                      </span>
                    </label>
                     
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          data-testid="CheckboxMonth11"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <span
                        class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
                      >
                        Nov
                      </span>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
